### PR TITLE
Fixed bug 3288

### PR DESCRIPTION
**Pull request by Dhruv-Aga**
This pull request addresses issue #3288 
This patch to handle the unexpected error raised by Entrez parser while parsing the newer document. Added Exception handler and removed assertion checks 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #...

### CODE REVIEW

File: Bio/Entrez/Parser.py [hunk 1]
```diff
@@ -434,8 +434,6 @@ def startNamespaceDeclHandler(self, prefix, uri):
             # to find out their first and last invocation for each namespace.
             self.namespace_level[prefix] += 1
             self.namespace_prefix[uri] = prefix
-            assert uri == "http://www.w3.org/1998/Math/MathML"
```
Comment (mdehoon) [inline Bio/Entrez/Parser.py:437]: The uri is either "http://www.w3.org/1998/Math/MathML" or "http://www.w3.org/1999/xlink", isn't it? Then we can keep the assert if we allow both values.

File: Bio/Entrez/Parser.py [hunk 2]
```diff
@@ -434,8 +434,6 @@ def startNamespaceDeclHandler(self, prefix, uri):
...
-            assert prefix == "mml"
```
Comment (mdehoon) [inline Bio/Entrez/Parser.py:438]: I guess this is still true in both cases?

File: Bio/Entrez/Parser.py (hunk 1 above)
Comment (Dhruv-Aga) [reply Bio/Entrez/Parser.py:437]: I think we can, actually my though was if URI is not from both the mentioned uri then also we should able to parse atleast some information.

Comment (Dhruv-Aga) [reply Bio/Entrez/Parser.py:437]: I think we can, actually my though was if URI is not from both the mentioned uri then also we should able to parse atleast some information.

File: Bio/Entrez/Parser.py (hunk 2 above)
Comment (Dhruv-Aga) [reply Bio/Entrez/Parser.py:438]: No, the case i tried i find that these are creating an unhandled error and stoping the parsing. Please take a look at this [example code](https://gist.github.com/Dhruv-Aga/c6f6f47cf4fd5f7a7bad78622f5a4709#file-issue-code-py) mentioned in  #3288. If there is any better solution, please guide me I will update my Pull request.

Comment (mdehoon) [reply Bio/Entrez/Parser.py:438]: I would suggest then to use
```python
    def startNamespaceDeclHandler(self, prefix, uri):
        """Handle start of an XML namespace declaration."""
        if prefix == "xsi":
            # This is an xml schema
            self.schema_namespace = uri
            self.parser.StartElementHandler = self.schemaHandler
        else:
            # Note that the DTD for MathML and XLink specify a default attribute
            # that declares the namespace for each MathML or Xlink element. This
            # means that these elements in the XML have an invisible namespace
            # declaration that triggers a call to startNamespaceDeclHandler
            # and endNamespaceDeclHandler. Therefore we need to count how often
            # startNamespaceDeclHandler and endNamespaceDeclHandler were called
            # to find out their first and last invocation for each namespace.
            if prefix == "mml":
                assert uri == "http://www.w3.org/1998/Math/MathML"
            elif prefix == "xlink":
                assert uri == "http://www.w3.org/1999/xlink"
            self.namespace_level[prefix] += 1
            self.namespace_prefix[uri] = prefix
        else:
            raise ValueError("Unknown prefix '%s' with uri %s" % (prefix, uri))

```

**Comment by mdehoon** [comment]
It looks like this will be difficult to fix in a clean way. The DTD file included with Biopython is from 2009. In the meantime, NCBI has updated this DTD; see the link https://dtd.nlm.nih.gov/ncbi/pmc/articleset/nlm-articleset-2.0.dtd shown at the top of the XML returned by Entrez. Unfortunately the new DTD is not very informative: Most elements are defined to be of type "ANY", which makes it impossible for the parser to figure out what should be a list, what should be a dictionary, what should be a string, etc.

I suggest then that we keep the old DTD from 2009, which allows the parser to handle most of the information in the XML.

File: Bio/Entrez/Parser.py (hunk 2 above)
Comment (Dhruv-Aga) [reply Bio/Entrez/Parser.py:438]: Thanks, I will update my Pull Request

**Comment by Dhruv-Aga** [comment]
Okay, So for new DTD format, i believe we should introduce a new feature or patch in long run to parse such document.

**Comment by mdehoon** [comment]
Thank you, it's merged.

**Pull request merged**